chalicelib/checks/wfr_checks.py

```diff
@@ -1,1449 +1,1436 @@
+            meta_workflow_uuid = meta_workflow_properties.get("uuid")
+            msg = "MetaWorkflow found: %s" % meta_workflow_name
+            check.brief_output.append(msg)
+            check.full_output["meta_workflow"] = meta_workflow_uuid
         else:
-            check.full_output['report_only'].append(info_pack)
-            action_category = 'Only Reported'
-        # add a short description for brief output
-        check.brief_output.append("{}, {}, {}, {}".format(a_wfr['uuid'],
-                                                          run_type,
-                                                          category,
-                                                          action_category
-                                                          ))
-
-    if check.full_output['cleanup']:
-        check.allow_action = True
-
-    report_catories = [i for i in category_dictionary if i not in delete_categories]
-    check.summary = "{} wfrs ({}) will be deleted, and {} wfrs ({}) are reported".format(
-        sum([counter[i] for i in delete_categories]),
-        ",".join([i for i in delete_categories if counter[i]]),
-        sum([counter[i] for i in report_catories]),
-        ",".join([i for i in report_catories if counter[i]])
+            msg = "MetaWorkflow not found: %s" % meta_workflow
+            check.brief_output.append(msg)
+    if cases:
+        cases_found = []
+        cases_without_sample_processing = []
+        cases = format_kwarg_list(cases)
+        found, cases_not_found = validate_items_existence(cases, connection)
+        for case in found:
+            sample_processing = case.get("sample_processing")
+            case_uuid = case.get("uuid")
+            cases_found.append(case_uuid)
+            if sample_processing:
+                sample_processings_from_cases.add(sample_processing)
+            else:
+                cases_without_sample_processing.append(case_uuid)
+        msg = "%s Case(s) found" % len(cases_found)
+        check.brief_output.append(msg)
+        check.full_output["cases_found"] = cases_found
+        if cases_not_found:
+            msg = "%s Case(s) not found" % len(cases_not_found)
+            check.brief_output.append(msg)
+            check.full_output["cases_not_found"] = cases_not_found
+        if cases_without_sample_processing:
+            msg = "%s Case(s) lacked a SampleProcessing" % len(
+                cases_without_sample_processing
+            )
+            check.brief_output.append(msg)
+            check.full_output[
+                "cases_without_sample_processing"
+            ] = cases_without_sample_processing
+    if sample_processings or sample_processings_from_cases:
+        sample_processings_found = []
+        sample_processings = format_kwarg_list(sample_processings)
+        sample_processings |= sample_processings_from_cases
+        found, not_found = validate_items_existence(sample_processings, connection)
```

Comment:
Another pattern repeated all over the place. In general a lot of these checks still have repeated patterns that may benefit from helper functions.